DatadogInternal/Sources/Utils/WithErrorContext.swift

```diff
@@ -0,0 +1,65 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-Present Datadog, Inc.
+ */
+
+import Foundation
+
+/// Adds contextual information to an error and rethrows it.
+///
+/// - Parameters:
+///   - fileID: The identifier of the source file where the error occurred. Default is `#fileID`.
+///   - line: The line number where the error occurred. Default is `#line`.
+///   - context: The contextual information to be added to the error. This will be evaluated lazily.
+///   - block: A closure that may throw an error.
+///
+/// - Throws: An `EnrichedError` containing the original error and the provided context.
+///
+/// - Returns: The result of the closure if no error is thrown.
+public func withErrorContext<T>(
+    fileID: StaticString = #fileID,
+    line: UInt = #line,
+    _ context: @autoclosure () -> String,
+    _ block: () throws -> T
+) throws -> T {
```

Comment:
I'm not a fan of this TBH, but I see the point. I wonder if we could have an `InternalError` extension or something 🤔

Reply:
Also, the function should `rethrows` instead.

Reply:
Removed 👍 as now we encode less blocks in `DS`. TLV errors are now enriched manually.